src/panels/PanelBase.js

```diff
@@ -97,41 +97,45 @@ export default {
   },
   methods: {
     toggle() {
-      this.localExpanded = !this.localExpanded;
-      if (this.localExpanded) {
-        this.isCached = true;
-      }
-    },
-    expand() {
-      this.localExpanded = !this.localExpanded;
-      if (this.localExpanded) {
+      if (!this.isCached && !this.localExpanded) {
+        // Let the content load before toggling localExpanded which triggers the animation,
+        // as we need to know the scrollHeight of the content for the animation to work.
         this.isCached = true;
```

Comment:
> So it seems like setting isCached here is actually our way of getting the panel to load the retriever and fetch the data. 

Not necessarily, it may be loaded from the `preload` attribute as well. ( has no effect on the code executed though )

> Should we rename this to something more accurate then? Maybe something like fetchSrc?

Hmm, I think the old method name `toggle` fits more, since it is used both when expanding / collapsing the panel.

Reply:
> Hmm, I think the old method name toggle fits more, since it is used both when expanding / collapsing the panel.

Sorry if I'm just being confused here, but do you mean rename `isCached` -> `toggle`? 

> Not necessarily, it may be loaded from the preload attribute as well. ( has no effect on the code executed though )

Please correct me if I'm wrong, but is it right to say that at the time we are setting `isCached = true`, we haven't actually loaded the data yet (assuming `preload` is false)? Setting `isCached = true` is just to trigger the rendering of the retriever, which then does the actual fetching/caching. 

If this is the case, would something like `fetchTriggered` be more appropriate?

Reply:
Ahh my bad! I misunderstood this for renaming `toggle -> fetchSrc`. 
Will change it to `isCached` -> `wasRetrieverLoaded` then

> Setting isCached = true is just to trigger the rendering of the retriever, which then does the actual fetching/caching.

yup